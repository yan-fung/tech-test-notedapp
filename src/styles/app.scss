@import "./_main";

* {
  margin: 0;
}