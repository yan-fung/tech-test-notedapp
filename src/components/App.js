import { useState } from "react";
import NewsletterSignup from "./NewsletterSignup";
import Reviews from "./Reviews";
import "../styles/app.css";

const App = () => {
  const [reviews, setReviews] = useState([]);
  const [isLoading, setIsLoading] = useState(true);
  return (
    <>
      <div className="App">
        <NewsletterSignup />
        <Reviews
          reviews={reviews}
          setReviews={setReviews}
          isLoading={isLoading}
          setIsLoading={setIsLoading}
        />
      </div>
    </>
  );
};

export default App;
