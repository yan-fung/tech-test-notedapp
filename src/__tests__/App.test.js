import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import App from "../components/App";
import "../styles/app.css";

it("renders app", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
