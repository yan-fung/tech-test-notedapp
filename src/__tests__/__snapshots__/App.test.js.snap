// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="subscription-container"
    >
      <div>
         Stay up-to-date with
      </div>
      <p>
        News, updates, promos, and exclusives from 
        <span
          class="noted"
        >
          Noted.
        </span>
      </p>
      <div
        class="subscription-form"
      >
        <form
          class="form"
        >
          <label
            class="label"
          >
            <input
              class="input"
              placeholder="Enter your email"
              type="email"
              value=""
            />
          </label>
          <button
            class="button"
            type="submit"
          >
            Subscribe
          </button>
        </form>
      </div>
    </div>
    <div
      class="review-content"
      data-testid="review__content"
    >
      <div>
        Loading...
      </div>
    </div>
  </div>
</DocumentFragment>
`;
